app/routes/authentication.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException, Path, Query
+from fastapi.security import OAuth2PasswordRequestForm
+
+from ..authentication.api_keys import api_key_is_valid
+from ..authentication.token import get_user_id, is_admin
+from ..crud import api_keys
+from ..errors import RecordNotFoundError, UnauthorizedError
+from ..models.orm.api_keys import ApiKey as ORMApiKey
+from ..models.pydantic.authentication import (
+    ApiKey,
+    APIKeyRequestIn,
+    ApiKeyResponse,
+    ApiKeysResponse,
+    ApiKeyValidation,
+    ApiKeyValidationResponse,
+    SignUpRequestIn,
+    SignUpResponse,
+)
+from ..utils.rw_api import login, signup
+
+router = APIRouter()
+
+
+@router.post("sign-up")
+async def sign_up(request: SignUpRequestIn):
+    data = await signup(request.name, request.email)
+    return SignUpResponse(data=data)
+
+
+@router.post("/token", tags=["Authentication"])
+async def get_token(form_data: OAuth2PasswordRequestForm = Depends()):
+    """Get access token from RW API."""
+    try:
+        token = await login(form_data.username, form_data.password)
+    except UnauthorizedError as e:
+        raise HTTPException(status_code=401, detail=str(e))
+
+    else:
+        return {
+            "access_token": token,
+            "token_type": "bearer",
+        }
+
+
+@router.post("/apikey", tags=["Authentication"], status_code=201)
+async def create_apikey(
+    request: APIKeyRequestIn,
+    user_id: str = Depends(get_user_id),
+):
+    """Request a new API key.
+
+    Default keys are valid for one year
+    """
+
+    input_data = request.dict(exclude_none=True, by_alias=True)
+
+    row: ORMApiKey = await api_keys.create_api_key(user_id=user_id, **input_data)
+
+    return ApiKeyResponse(data=row)
+
+
+@router.get("/apikeys", tags=["Authentication"])
```

Comment:
same as above, should it be 'api_keys' instead of 'apikeys'?

Reply:
test_v2 will replace test at one point. This will be a rewrite of all tests to improve test coverage and reduce test runtime. F